### PR TITLE
Adding aliases when creating index.

### PR DESCRIPTION
**Pull request by jorjika**
This addreses issue #289 
It does not introduce breaking change.

Adding aliases when creating index.
Changes in Serilog.Sinks.Elasticsearch

### CODE REVIEW

**Comment by jorjika** [comment]
Omitted to update template_v7.json in Tests

File: sample/Serilog.Sinks.Elasticsearch.Sample/Program.cs [hunk 1]
```diff
@@ -26,24 +26,55 @@ class Program
...
-                                })
-                    .CreateLogger();
+
+
+
+            Log.Logger.Information("Hi");
+            Log.Information("Hello, world!");
+            Console.ReadLine();
+
+            //Log.Logger = new LoggerConfiguration()
```
Comment (mivano) [inline sample/Serilog.Sinks.Elasticsearch.Sample/Program.cs]: Lot of commented out code. Either remove or give it a purpose?

File: sample/Serilog.Sinks.Elasticsearch.Sample/Program.cs [hunk 2]
```diff
@@ -26,24 +26,55 @@ class Program
...
+            Log.Logger = new LoggerConfiguration()
+               //.Enrich.With<MachineNameEnricher>()
+               .Enrich.WithProperty("Version", "v1")
+               .Enrich.WithProperty("Application", "YourAppName")
+               .Enrich.WithProperty("HostName", Environment.MachineName)
+               .Enrich
+               .FromLogContext()
+               .MinimumLevel.Warning()
+               .MinimumLevel.Override("Serilog", LogEventLevel.Information)
+               .MinimumLevel.Override("JorJika", LogEventLevel.Information)
```
Comment (mivano) [inline sample/Serilog.Sinks.Elasticsearch.Sample/Program.cs]: Why is this needed?

File: src/Serilog.Sinks.Elasticsearch/LoggerConfigurationElasticSearchExtensions.cs [hunk 3]
```diff
@@ -170,6 +171,7 @@ public static class LoggerConfigurationElasticsearchExtensions
             string deadLetterIndexName = null,
             int? numberOfShards = null,
             int? numberOfReplicas = null,
+            string[] indexAliases = null,
```
Comment (mivano) [inline src/Serilog.Sinks.Elasticsearch/LoggerConfigurationElasticSearchExtensions.cs]: Adding a new parameter is a breaking change as it changes the signature. Do you really need this here and not just as a property on the options object?

File: src/Serilog.Sinks.Elasticsearch/Sinks/ElasticSearch/ElasticsearchSinkOptions.cs [hunk 4]
```diff
@@ -76,6 +76,14 @@ public class ElasticsearchSinkOptions
         /// </summary>
         public int? NumberOfReplicas { get; set; }
 
+        /// <summary>
+        /// Index aliases. Used in rollover jobs and etc.
```
Comment (mivano) [inline src/Serilog.Sinks.Elasticsearch/Sinks/ElasticSearch/ElasticsearchSinkOptions.cs]: Can you elaborate what happens when you set the value?

File: test/Serilog.Sinks.Elasticsearch.Tests/Templating/template_v7.json [hunk 5]
```diff
@@ -71,5 +71,6 @@
                 }
             }
         }
-    }
+    },
+    "aliases": {}
```
Comment (mivano) [inline test/Serilog.Sinks.Elasticsearch.Tests/Templating/template_v7.json]: Can you also add a test that shows that it indeed sets the aliases when provided?

**Comment by mivano** [comment]
Thanks for the PR. I added some small comments.

File: sample/Serilog.Sinks.Elasticsearch.Sample/Program.cs (hunk 1 above)
Comment (jorjika) [reply sample/Serilog.Sinks.Elasticsearch.Sample/Program.cs]: I used sample project for testing so I commented some parts of code. Now i returned everything back as it was.

File: sample/Serilog.Sinks.Elasticsearch.Sample/Program.cs (hunk 2 above)
Comment (jorjika) [reply sample/Serilog.Sinks.Elasticsearch.Sample/Program.cs]: Removed.

File: src/Serilog.Sinks.Elasticsearch/LoggerConfigurationElasticSearchExtensions.cs (hunk 3 above)
Comment (jorjika) [reply src/Serilog.Sinks.Elasticsearch/LoggerConfigurationElasticSearchExtensions.cs]: Removed from there. You are right, its acceptable to assign value later, not from that extension method.

File: src/Serilog.Sinks.Elasticsearch/Sinks/ElasticSearch/ElasticsearchSinkOptions.cs (hunk 4 above)
Comment (jorjika) [reply src/Serilog.Sinks.Elasticsearch/Sinks/ElasticSearch/ElasticsearchSinkOptions.cs]: I've changed comment. Index alias is the level property as NumberOfReplicas. Hope this comment is enough :).

File: test/Serilog.Sinks.Elasticsearch.Tests/Templating/template_v7.json (hunk 5 above)
Comment (jorjika) [reply test/Serilog.Sinks.Elasticsearch.Tests/Templating/template_v7.json]: I've added one alias in test and also changed template_v7.json.

**Comment by mivano** [comment]
Thanks for the changes, sorry that it took some time to merge. A pre-release package will be created, please see if this suites your needs.
